pylops/waveeqprocessing/kirchhoff.py

```diff
@@ -406,7 +409,27 @@ def __init__(
         self._register_multiplications(engine)
 
     @staticmethod
-    def _identify_geometry(z, x, srcs, recs, y=None):
+    def _identify_geometry(
+        z: NDArray,
+        x: NDArray,
+        srcs: NDArray,
+        recs: NDArray,
+        y: Optional[NDArray] = None,
+    ) -> Union[
```

Comment:
should be `Tuple`

Reply:
I assume you mean the return right?

Reply:
If this is the case there are a few other cases, eg `Marchenko.apply_onepoint` where this should be fixed

Reply:
Those are `apply_onepoint` are fine because they can actually output different things based on the input. So a `Union` is justified. If you are always returning the same type of output, and that output is composed of many components, choose a Tuple. If you are outputting different outputs depending on the call, use a Union.